pkg/retry/retry_test.go

```diff
@@ -0,0 +1,211 @@
+			},
+			githubOverride: &MockGitHub{
+				listDeliveries: &listDeliveriesRes{
+					deliveries: []*github.HookDelivery{
+						{
+							StatusCode: createInt(http.StatusOK),
+						},
+					},
+					res: &github.Response{},
+				},
+			},
+		},
+		{
+			name:          "github_list_deliveries_failure",
+			expStatusCode: http.StatusInternalServerError,
+			expRespBody:   http.StatusText(http.StatusInternalServerError),
+			datastoreClientOverride: &MockDatastore{
+				retrieveCheckpointID: &retrieveCheckpointIDRes{res: "checkpoint-id"},
+			},
+			gcsLockClientOverride: &MockLock{
+				acquire: &acquireRes{},
+			},
+			githubOverride: &MockGitHub{
+				listDeliveries: &listDeliveriesRes{err: errors.New("error")},
+			},
+		},
+		{
+			name:          "github_redeliver_event_failure",
+			expStatusCode: http.StatusInternalServerError,
+			expRespBody:   http.StatusText(http.StatusInternalServerError),
+			datastoreClientOverride: &MockDatastore{
+				retrieveCheckpointID: &retrieveCheckpointIDRes{res: "checkpoint-id"},
+			},
+			gcsLockClientOverride: &MockLock{
+				acquire: &acquireRes{},
+			},
+			githubOverride: &MockGitHub{
+				listDeliveries: &listDeliveriesRes{
+					deliveries: []*github.HookDelivery{
+						{
+							ID:         createInt64(1),
+							StatusCode: createInt(http.StatusInternalServerError),
+						},
+					},
+					res: &github.Response{},
+				},
+				redeliverEvent: &redeliverEventRes{err: errors.New("error")},
+			},
+		},
+		{
+			name:          "success",
+			expStatusCode: http.StatusAccepted,
+			expRespBody:   http.StatusText(http.StatusAccepted),
+			datastoreClientOverride: &MockDatastore{
+				retrieveCheckpointID: &retrieveCheckpointIDRes{res: "checkpoint-id"},
+			},
+			gcsLockClientOverride: &MockLock{
+				acquire: &acquireRes{},
+			},
+			githubOverride: &MockGitHub{
+				listDeliveries: &listDeliveriesRes{
+					deliveries: []*github.HookDelivery{
+						{
+							StatusCode: createInt(http.StatusOK),
+						},
+					},
+					res: &github.Response{},
+				},
+			},
+		},
+	}
+
+	for _, tc := range cases {
+		tc := tc
+
+		t.Run(tc.name, func(t *testing.T) {
+			t.Parallel()
+
+			rco := &RetryClientOptions{
+				DatastoreClientOverride: tc.datastoreClientOverride,
+				GCSLockClientOverride:   tc.gcsLockClientOverride,
+				GitHubOverride:          tc.githubOverride,
+			}
+
+			srv, err := NewServer(ctx, &Config{}, rco)
+			if err != nil {
+				t.Fatalf("failed to create new server: %v", err)
+			}
+
+			var payload []byte
+			req := httptest.NewRequest(http.MethodPost, "/retry", bytes.NewReader(payload))
+
+			resp := httptest.NewRecorder()
+
+			srv.handleRetry().ServeHTTP(resp, req)
+
+			if resp.Code != tc.expStatusCode {
+				t.Errorf("StatusCode got: %d want: %d", resp.Code, tc.expStatusCode)
+			}
+
+			if resp.Body.String() != tc.expRespBody {
+				t.Errorf("ResponseBody got: %s want: %s", resp.Body.String(), tc.expRespBody)
+			}
+		})
+	}
+}
+
+func createInt64(x int) *int64 {
```

Comment:
These feel unnecessary to me. Why have a function to do simple conversions instead of just inlining it?

Reply:
It'd be more common to name this `int64Ptr` and `intPtr`. With generics, you can create `toPtr`:

```go
func toPtr[T any](i T) *T {
  return &i
}
```